frontend/packages/multicluster-sdk/src/components/FleetResourceLink.tsx

```diff
@@ -0,0 +1,253 @@
+ * multi-cluster resources.
+ *
+ * @see https://github.com/openshift/console/blob/main/frontend/packages/console-dynamic-plugin-sdk/docs/api.md#resourcelink
+ *
+ * @param props - FleetResourceLinkProps extending ResourceLinkProps with cluster information
+ * @param props.cluster - the target cluster name for the resource
+ * @param props.groupVersionKind - K8s GroupVersionKind for the resource
+ * @param props.name - the resource name
+ * @param props.namespace - the resource namespace (required for namespaced resources)
+ * @param props.displayName - optional display name override
+ * @param props.className - additional CSS classes
+ * @param props.inline - whether to display inline
+ * @param props.hideIcon - whether to hide the resource icon
+ * @param props.children - additional content to render
+ *
+ * @example
+ * ```typescript
+ * // Hub cluster VirtualMachine - routes to ACM VM page on multicloud paths
+ * <FleetResourceLink
+ *   name="my-vm"
+ *   namespace="default"
+ *   groupVersionKind={{ group: 'kubevirt.io', version: 'v1', kind: 'VirtualMachine' }}
+ * />
+ *
+ * // Managed cluster VirtualMachine - routes to ACM search results
+ * <FleetResourceLink
+ *   name="remote-vm"
+ *   namespace="default"
+ *   cluster="prod-cluster"
+ *   groupVersionKind={{ group: 'kubevirt.io', version: 'v1', kind: 'VirtualMachine' }}
+ * />
+ *
+ * // ManagedCluster resource (lives on hub) - cluster prop omitted
+ * <FleetResourceLink
+ *   name="prod-cluster"
+ *   groupVersionKind={{ group: 'cluster.open-cluster-management.io', version: 'v1', kind: 'ManagedCluster' }}
+ * />
+ * ```
+ */
+export const FleetResourceLink: React.FC<FleetResourceLinkProps> = ({ cluster, ...resourceLinkProps }) => {
+  const [hubClusterName, hubLoaded] = useHubClusterName()
+  const location = useLocation()
+
+  // hook that handles useResolvedExtensions and lookup logic
+  const { resourceRoutesResolved, getResourceRouteHandler } = useResourceRouteExtensions()
+
+  // check if fleet is available (has managed clusters)
+  const isFleetAvailable = hubLoaded
+
+  if (!isFleetAvailable) {
+    // fallback to default ResourceLink from OCP
+    return <ResourceLink {...resourceLinkProps} />
+  }
+
+  const {
+    className,
+    displayName,
+    inline = false,
+    groupVersionKind,
+    name,
+    nameSuffix,
+    namespace,
+    hideIcon,
+    title,
+    children,
+    dataTest,
+    onClick,
+    truncate,
+  } = resourceLinkProps
+
+  const value = displayName || name
+  const classes = classNames('co-resource-item', className || '', {
+    'co-resource-item--inline': inline,
+    'co-resource-item--truncate': truncate,
+  })
+
+  // if cluster name is given but hub name is not loaded yet, show skeleton
+  if (cluster && !hubLoaded) {
+    return (
+      <span className={classes}>
+        {!hideIcon && <ResourceIcon groupVersionKind={groupVersionKind} />}
+        <Skeleton width="100px" />
```

Comment:
I'm not sure about the fixed width here... since we know the name of the item, just not where it should link to yet, perhaps we should just show it as text in this case, rather than a skeleton? That would avoid hardcoding a width.